projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -259,30 +263,23 @@ public function set_site_card_menu_class( array $menu ) {
 	}
 
 	/**
-	 * Returns the first available upsell nudge.
+	 * Enqueue the script to async load the upsell nudge.
+	 */
+	public function enqueue_async_upsell_nudge() {
+		wp_enqueue_script( 'admin-menu-upsell-nudge', plugins_url( 'upsell-nudge.js', __FILE__ ), array( 'wp-api-fetch' ), $this->cache_bust, true );
+	}
+
+	/**
+	 * Modify the async upsell nudge script tag to be async.
 	 *
-	 * @return array
+	 * @param string $tag    Prior script tag markup.
+	 * @param string $handle Name of the script handle we're going to load.
 	 */
-	public function get_upsell_nudge() {
-		$jitm         = \Automattic\Jetpack\JITMS\JITM::get_instance();
-		$message_path = 'calypso:sites:sidebar_notice';
-		$message      = $jitm->get_messages( $message_path, wp_json_encode( array( 'message_path' => $message_path ) ), false );
-
-		if ( isset( $message[0] ) ) {
-			$message = $message[0];
-			return array(
-				'content'                      => $message->content->message,
-				'cta'                          => $message->CTA->message, // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'link'                         => $message->CTA->link, // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'tracks_impression_event_name' => $message->tracks->display->name,
-				'tracks_impression_cta_name'   => $message->tracks->display->props->cta_name,
-				'tracks_click_event_name'      => $message->tracks->click->name,
-				'tracks_click_cta_name'        => $message->tracks->click->props->cta_name,
-				'dismissible'                  => $message->is_dismissible,
-				'feature_class'                => $message->feature_class,
-				'id'                           => $message->id,
-			);
+	public function async_script_loader_tag( $tag, $handle ) {
```

Comment:
I wonder if we could rely on `Assets::enqueue_async_script` for this script.

Reply:
I think I prefer what we have currently. It's also consistent with the rest of the code that we're using, but also because we currently load the `admin-menu.js` script the same way, so I prefer staying consistent. Porting both could be a good follow-up task, of course.

Reply:
I've gone and changed `async` to `defer` in https://github.com/Automattic/jetpack/pull/24222/commits/fa603cb4e7b2e823a4c016a663e3313c98b40ed9 though, as this was something I intended to do earlier.

Reply:
> we currently load the `admin-menu.js` script the same way, so I prefer staying consistent. Porting both could be a good follow-up task, of course.

I might be missing something, but AFAIC the `admin-menu.js` script is not a deferred script and thus not loaded the same way nor can use `Assets::enqueue_async_script`. 

Said that, I agree with @jeherve above. I find less confusing to use `Assets::enqueue_async_script` for loading deferred scripts (since this `async_script_loader_tag` function is esentially duplicating [an existing logic](https://github.com/Automattic/jetpack/blob/dacba4eaac61bbada8fed9530f704beaf9322d67/projects/packages/assets/src/class-assets.php#L89)) and leave `wp_enqueue_script` for non-deferred scripts.

Anyway, I don't consider this a blocker though 👍

Reply:
> use `Assets::enqueue_async_script` for loading deferred scripts

On a second thought, I don't think that's available in `wpcom` for simple sites, so it might be preferable to leave things as they are in order to have the same logic in both  `Atomic_Admin_Menu` and `WPcom_Admin_Menu`.